x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/register.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CollectorFetchContext, UsageCollectionSetup } from '@kbn/usage-collection-plugin/server';
+import { getIndicesStats } from './indices_stats_collector';
+import { cspmUsageSchema } from './schema';
+import { CspmUsage } from './types';
+
+export function registerCspmUsageCollector(usageCollection?: UsageCollectionSetup): void {
+  // usageCollection is an optional dependency, so make sure to return if it is not registered.
+  if (!usageCollection) {
+    return;
+  }
+
+  // create usage collector
+  const indicesCounterCollector = usageCollection.makeUsageCollector<CspmUsage>({
```

Comment:
nit: naming is a bit off - it is `cspmUsageCollector` - not only indices. `indices` is one property in that collection